### ISSUE TITLE
Intersphinx links do not work for command-line options

### ISSUE THREAD

**Issue by brechtm** [sphinx-doc/sphinx#2822]
With 

```
intersphinx_mapping = {'sphinx': ('http://www.sphinx-doc.org/en/1.4.5', None)}
```

set in my project's `conf.py`,

```
:option:`sphinx:sphinx-build -b`
```

fails with

```
WARNING: unknown option: sphinx:sphinx-build -b
```

The following also fail with a similar warning:

```
 :option:`sphinx-build -b`
 :option:`-b`
 :option:`sphinx:-b`
```

These work fine, however:

```
:class:`sphinx:sphinx.application.Sphinx`
:confval:`latex_documents`
```

**Comment by Paebbels** [comment]
Any updates on this issue?

I have a similar problem to reference command line option in an another repository called `ghdl`. The inventory file was downloaded and looks fine, but Sphinx creates an unknown option warning.

Documentation URL: http://ghdl.readthedocs.io/en/latest/
Example option: `--ieee`

---

Related issues:
* tgingold/ghdl#200
* sphinx-doc/sphinx#3487

**Comment by tk0miya** [comment]
Thank you for reporting.
Fixed at 811df6b.

**Comment by chipx86** [comment]
Have people confirmed that this bug is fixed? I'm running 1.6.1 and option references still don't work unless I use the form of:

```
:option:`--some-option <intersphinxset:my-program.--some-option>`
```

These do not work:

```
:option:`intersphinxset:my-program --some-option`
:option:`my-program --some-option`
```

And while perhaps not intended behavior, referencing options in another document by first defining a `.. program::` (as a shortcut for prefixing each option with the program name) works anywhere within a Sphinx doc set, but there's inconsistent behavior here when trying to reference options in another doc set via intersphinx:

```
.. program:: my-program

This does not work: :option:`intersphinxset:--some-option`
But this does: :option:`--some-option`
```

In the first case, the reference turns into `my-program.intersphinxset:--some-option`, which is a bit wrong. This seems to be due to `StandardDomain._resolve_option_xref` just not considering that there might be an intersphinx prefix when building strings.

Looking at the `objects.inv` file, I see entries like `my-program.--some-option`, but it doesn't appear that Sphinx is able to re-assemble that form when using `:option:`? Normally when defining options, `progoptions` data is generated that allows `:option:` to find the correct reference name. The problem appears to be that the `progoptions` data is not stored in the `objects.inv`, so these reference names can't be built correctly in another doc set when using intersphinx.

Am I missing something there? I'm still learning how the intersphinx inventory and domain data interact.
